packages/react-components/src/components/MessageThread.tsx

```diff
@@ -360,14 +378,107 @@ const memoizeAllMessages = memoizeFnAll(
       disableEditing
     };
 
+    /* @conditional-compile-remove(data-loss-prevention) */
```

Comment:
The logic IS the same, the only reason is because condition-compile-remove does not work for switch case, when merging to stable will share the same switch case with slight logic change

Reply:
Most likely we will only remove the beta tag then forget to do the cleanup for the code, so it is always to do it earlier than later

 Would be nice to extract this common part into an internal function, and yes, we could still come back later to make it neat, but let's try our best when it is possible

Reply:
Will refactor out common code